quarkus-container-image-s2i extension is throw an image capability error

## Describe the bug

When I run a simple test with an application that has a `quarkus-container-image-s2i ` extension, I got the following error:
```
Caused by: io.quarkus.builder.BuildException: 
Build failure: Build failed due to errors
        [error]: Build step io.quarkus.container.image.deployment.ContainerImageProcessor#publishImageInfo threw an exception: java.lang.IllegalArgumentException: Unknown container image capability: io.quarkus.container.image.s2i
```

I think that the error could be related to this PR: https://github.com/quarkusio/quarkus/pull/13960

**Quarkus Version**: Upstream and 2.0.0.Alpha1
**Note:** it worked in Quarkus 1.11
**Reproducer:**
            project: https://github.com/quarkus-qe/quarkus-openshift-test-suite/tree/main/http/http-minimum 
            cmd: `mvn -B -V clean verify -fae -Dversion.quarkus=999-SNAPSHOT -pl app-metadata/deployment,app-metadata/runtime,common,http/http-minimum -Dts.authenticated-registry`

--- a/core/deployment/src/main/java/io/quarkus/deployment/Capability.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/Capability.java
@@ -75,7 +75,7 @@ public interface Capability {
     String METRICS = QUARKUS_PREFIX + "metrics";
     String CONTAINER_IMAGE_JIB = QUARKUS_PREFIX + "container.image.jib";
     String CONTAINER_IMAGE_DOCKER = QUARKUS_PREFIX + "container.image.docker";
-    String CONTAINER_IMAGE_S2I = QUARKUS_PREFIX + "container.image.openshift";
+    String CONTAINER_IMAGE_S2I = QUARKUS_PREFIX + "container.image.s2i";
     String CONTAINER_IMAGE_OPENSHIFT = QUARKUS_PREFIX + "container.image.openshift";
     String HIBERNATE_ORM = QUARKUS_PREFIX + "hibernate.orm";
     String HIBERNATE_ENVERS = QUARKUS_PREFIX + "hibernate.envers";

